frontend/express/public/core/events/javascripts/countly.overview.views.js

```diff
@@ -220,6 +220,7 @@
                     return this.$store.getters["countlyEventsOverview/selectedDatePeriod"];
                 },
                 set: function(value) {
+                    countlyCommon.setPeriod(value);
```

Comment:
you can get rid of this.
and move this.$store.dispatch('countlyEventsOverview/fetchMonitorEvents'); to dateChanged method.